neurodamus/connection_manager.py

```diff
@@ -775,7 +776,13 @@ def _get_conn_stats(self, dst_target):
                 sample_len = len(sample)
                 if not sample_len:
                     continue
-                sample_counts = self._synapse_reader.get_counts(sample, group_by="syn_type_id")
+                try:
+                    sample_counts = self._synapse_reader.get_counts(sample, group_by="syn_type_id")
```

Comment:
`syn_type_id` is mandatory for "chemical" edges. But it is not required for other type of edges file, e.g. "electrical", the ngv edges. See the [SONATA SPEC](https://sonata-extension.readthedocs.io/en/latest/sonata_tech.html#edge-file). Is it possible to count with another (maybe common) attributes?

Reply:
Thanks for the comment Weina. 
The field is used for counting but also to discriminate between different types of synapses.
I guess that in the case of electrical synapses there's no need to count different types?

Reply:
You are right, for electrical synapses, there is only one type : the gap juction `Nd.Gap` .

Reply:
So we can make the assumption that if there's no `syn_type_id` we should count them as `Nd.Gap`?

Reply:
Maybe do like this: if calling `_get_conn_stats` from  `GapJunctionManager` class, we count all synapse as `Nd.Gap`. 
In the ngv case, the `astrocyte` edges don't  contain  the  attribute `syn_type_id` either, but there the creation of syn objects is different from the general case.